src/main/java/com/algorand/algosdk/templates/LimitOrder.java

```diff
@@ -1,4 +1,143 @@
+     * @param ratn the numerator of the exchange rate
+     * @param ratd the denominator of the exchange rate
+     * @param expirationRound the round on which the assets can be transferred back to owner
+     * @param minTrade the minimum amount (of Algos) to be traded away
+     * @param maxFee the maximum fee that can be paid to the network by the account
+     */
+    public static ContractTemplate MakeLimitOrder(final Address owner, final Integer assetId, final Integer ratn, final Integer ratd, final Integer expirationRound, final Integer minTrade, final Integer maxFee) throws NoSuchAlgorithmException {
+        Objects.requireNonNull(owner);
+        Objects.requireNonNull(assetId);
+        Objects.requireNonNull(ratn);
+        Objects.requireNonNull(ratd);
+        Objects.requireNonNull(expirationRound);
+        Objects.requireNonNull(minTrade);
+        Objects.requireNonNull(maxFee);
+
+        List<ParameterValue> values = ImmutableList.of(
+                new IntParameterValue(5, maxFee),
+                new IntParameterValue(7, minTrade),
+                new IntParameterValue(9, assetId),
+                new IntParameterValue(10, ratd),
+                new IntParameterValue(11, ratn),
+                new IntParameterValue(12, expirationRound),
+                new AddressParameterValue(16, owner)
+        );
+
+        byte[] program = Encoder.decodeFromBase64(referenceProgram);
+        return inject(program, values);
+    }
+
+    /**
+     * Creates a group transaction array which transfer funds according to the contract's ratio
+     *
+     * @param contract previously created LimitOrder contract
+     * @param assetAmount amount of assets to be sent
+     * @param sender account to sign the payment transaction
+     * @param feePerByte fee per byte used for the transactions
+     * @param algoAmount number of algos to transfer
+     * @param firstValid first round on which these txns will be valid
+     * @param lastValid last round on which these txns will be valid
+     * @param genesisHash Genesis hash for the network where the transaction will be submitted.
+     * @return
+     * @throws IOException
+     */
+    public static byte[] MakeSwapAssetsTransaction(ContractTemplate contract, Integer assetAmount, Account sender, Integer feePerByte, Integer algoAmount, Integer firstValid, Integer lastValid, Digest genesisHash) throws IOException, NoSuchAlgorithmException {
+        Logic.ProgramData data = readAndVerifyContract(contract.program, 10, 1);
+
+        Address owner = new Address(data.byteBlock.get(0));
+        int assetId = data.intBlock.get(6);
+        int ratd = data.intBlock.get(7);
+        int ratn = data.intBlock.get(8);
+
+        // Verify the exchange rate ratio
+        if (assetAmount * ratd < algoAmount * ratn) {
+            throw new IllegalArgumentException("The provided amounts to not meet the contract exchange rate of assetAmount = algoAmount * " + ratn + " / " + ratd);
+        }
+
```

Comment:
It might also be good to check for algoAmount > minTrade here since we read the contract anyway.